### ISSUE TITLE
Isolate Naas install from other python libraries 

### ISSUE THREAD

**Issue by jravenel** [jupyter-naas/naas#301]
We need to isolate Naas install from other python libraries to enable https://github.com/jupyter-naas/awesome-notebooks/issues/209 for example.

People could install python lib different from the one installed in Naas. 

FYI @riderx

**Comment by riderx** [comment]
yes same issue i got for Supabase, the goto seems to use virtual env :
https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/

**Comment by Dr0p42** [comment]
> yes same issue i got for Supabase, the goto seems to use virtual env : https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/

Yes @riderx we need to implement this it's in the roadmap. It will be easier for everyone I think 👍

**Comment by Dr0p42** [comment]
> Somthing like that `python -m pip install --upgrade pip virtualenv`

The thing is that we will install naas + naas dependencies in its own virtualenv, then before running naas we will activate that venv. It will then be isolated.

I already have a branch on that which I need to push.
